feature-libs/product-configurator/rulebased/components/group-menu/configurator-group-menu.component.service.ts

```diff
@@ -0,0 +1,112 @@
+import { Inject, Injectable, PLATFORM_ID } from '@angular/core';
+import { DOCUMENT, isPlatformBrowser } from '@angular/common';
+import { BREAKPOINT, BreakpointService } from '@spartacus/storefront';
+import { take } from 'rxjs/operators';
+
+@Injectable({ providedIn: 'root' })
+export class ConfiguratorGroupMenuService {
+  constructor(
+    @Inject(PLATFORM_ID) protected platformId: any,
+    @Inject(DOCUMENT) protected document,
+    protected breakpointService?: BreakpointService
```

Comment:
Why this dependency is optional? Typo?

Reply:
Update: I hope we can get rid of this service at all, see comment: https://github.com/SAP/spartacus/pull/12348#discussion_r638070838

Reply:
Good hint!  I removed `?` after breakpointService